### ISSUE TITLE
Taln Hackathon

### ISSUE THREAD

**Issue by joancf** [openminted/Open-Call-Discussions#29]
Mostly I would like to know how to ensure that our components actually work on the platform. We  have shared a component as docker but i need some xmi files to test it. and to know the way to run it...

Thanks

Joan

**Comment by greenwoodma** [comment]
To make sure we are as well prepared as possible to help during the hackathon sessions could you please add/attach to this issue:

1. The landing page URL of any component/workflow you have registered
2. The OMTD-SHARE XML file for each component/workflow
3. One or two sample documents that you expect to produce sensible output for your component/workflow

**Comment by gkirtzou** [comment]
@joancf  Please provide as soon as possible the information requested byu @greenwoodma  so that we could check the metadata and verify that your components/application can work in the OMTD platform.

**Comment by joancf** [comment]
Yes, 
Our components are two dockers
1 Freeling
can be found it 
hub.doker    taln/openminted_freeling
the OMTD-Share XML 
https://github.com/TalnUPF/OpenMinted_Freeling/blob/master/OpenMinted_Descriptor.xml

2 Babelnet
 can be found it 
hub.doker    taln/openminted_babelnet
the OMTD-Share XML 
https://github.com/TalnUPF/OpenMinted_BabelNet/blob/master/OpenMinted_Descriptor.xml

I will attach the input files before tomorrow (sorry I have some, but now I don't have the computer where they are stored)

**Comment by gkirtzou** [comment]
As far as the Freeling component is concerned 
(1) 
In distributionInfos>, change the distribution location just to _taln/openminted_freeling:version0_, where version is the tagged version of the component you are registring. It is recommened to add a version and not to use the latest for reproducibility reasons.

(2)
In distributionInfo, change the command to "openminted_freeling process.sh" , or "process.sh" (I am not exactly which is the correct one), and verify that whichever command you call that it is callable from everywhere in the docker container. See also [here](https://github.com/openminted/Open-Call-Discussions/issues/28#issuecomment-381683553). 
Note that the OMTD platform will generate a command similar to the one you provided initially, ie.
_docker run -v /var/data/openminted/input:/input -ti -v /var/data/openminted/output:/output openminted_freeling process.sh --input /input --output /output --param:language=en_, taking into the account that we run a docker image via the galaxy workflow engine. 

(3) 
In  parameterInfo for the parameter "Language", please add a default value as it is obligatory. It would be really helpful when one tries to use it.

(4) 
In inputContentResourceInfo, you have declared the processingResourceType as document. Don't your component work on a corpus of documents? If that's the case, please change it to corpus. 

(5) 
In inputContentResourceInfo, you have declared the following data formats:

< ns0:dataFormats>
      < ns0:dataFormatInfo>
               < ns0:dataFormat>http://w3id.org/meta-share/omtd-share/UimaCasFormat</ns0:dataFormat>
                < ns0:dataFormatOther>XMI</ns0:dataFormatOther>
      </ns0:dataFormatInfo>
 </ns0:dataFormats>

I think i would be more consistent to declare multiple data formats both for UICasFormat ([http://w3id.org/meta-share/omtd-share/UimaCasFormat](http://w3id.org/meta-share/omtd-share/UimaCasFormat)), as well as for the XMI ([http://w3id.org/meta-share/omtd-share/Xmi](http://w3id.org/meta-share/omtd-share/Xmi))

(6) 
In  inputContentResourceInfo, you have declared the following data format:
< ns0:annotationType>http://w3id.org/meta-share/omtd-share/StructuralAnnotationType</ns0:annotationType>
That means that you required as input something that is already annotated at that level, correct?

(7) 
For outputResourceInfo, similar comments apply as in (4) and (5) 

Please, if you update the OMTD-SHARE descriptor according to my comments, send them back to check them again :)

**Comment by gkirtzou** [comment]
As far as the Babelnet component is concerned, 
(1) , (2) , (4), (5) from the comment above also apply. 

Also,

- In inputContentResouceInfo, you declared as dataformat http://w3id.org/meta-share/omtd-share/Token</ns0:annotationType and http://w3id.org/meta-share/omtd-share/PartOfSpeech. That means that the input is something already annotated at that level, correct?
- Is your output a single document for the whole corpus with the extracted terms or annotations of the terms within the corpus?

**Comment by joancf** [comment]
Thanks Katerina, I'll do it ASAP.

About the Freeling component , I have some doubts on your comments , please find them inlined:



On 04/18/2018 01:01 PM, Katerina Gkirtzou wrote:
>
> As far as the Freeling component is concerned
> (1)
> In distributionInfos>, change the distribution location just to taln/openminted_freeling:version0, where version is the tagged version of the component you are registring. It is recommened to add a version and not to use the latest for reproducibility reasons.

ok
>
> (2)
> In distributionInfo, change the command to "openminted_freeling process.sh" , or "process.sh" (I am not exactly which is the correct one), and verify that whichever command you call that it is callable from everywhere in the docker container. See also here.
> Note that the OMTD platform will generate a command similar to the one you provided initially, ie.
> docker run -v /var/data/openminted/input:/input -ti -v /var/data/openminted/output:/output openminted_freeling process.sh --input /input --output /output --param:language=en, taking into the account that we run a docker image via the galaxy workflow engine.

ok, i'll change it, is in /bin so it should work.
>
> (3)
> In parameterInfo for the parameter "Language", please add a default value as it is obligatory. It would be really helpful when one tries to use it.

ok, I will put the "auto" value, it is not the most efficient but the most robust one.
But, the default value is a value that  do you provide, or I must also consider that maybe you don't specify any value and I have to deal with that?
>
> (4)
> In inputContentResourceInfo, you have declared the processingResourceType as document. Don't your component work on a corpus of documents? If that's the case, please change it to corpus.

That was a doubt I got, when filling the form, because I understood that a corpus is a set of documents that the component processes as a unit (for example to make a mult-document summary).  In our case the Freeling component processes documents in a folder but does not consider them a unit. But I will change that
>
> (5)
> In inputContentResourceInfo, you have declared the following data formats:
>
> < ns0:dataFormats>
> < ns0:dataFormatInfo>
> < ns0:dataFormat>http://w3id.org/meta-share/omtd-share/UimaCasFormat</ns0:dataFormat>
> < ns0:dataFormatOther>XMI</ns0:dataFormatOther>
> </ns0:dataFormatInfo>
> </ns0:dataFormats>
>
> I think i would be more consistent to declare multiple data formats both for UICasFormat (http://w3id.org/meta-share/omtd-share/UimaCasFormat), as well as for the XMI (http://w3id.org/meta-share/omtd-share/Xmi)

I got many doubts on that. You don't have the XMI format!! and no instructions on how to express it So, my input/output are xmi files (and the typesystem.xml file) should I just indicate xml? Please if you can tell me what I have to put there I will just copy your anser.
>
> (6)
> In inputContentResourceInfo, you have declared the following data format:
> < ns0:annotationType>http://w3id.org/meta-share/omtd-share/StructuralAnnotationType</ns0:annotationType>
> That means that you required as input something that is already annotated at that level, correct?

In that case I only need text, I don't need nothing else, but once I  clicked the option there was no way to remove it. I think it should be empty
>
> (7)
> For outputResourceInfo, similar comments apply as in (4) and (5)
>
> Please, if you update the OMTD-SHARE descriptor according to my comments, send them back to check them again :)

**Comment by joancf** [comment]
About the Babelnet component
>As far as the Babelnet component is concerned,
>(1) , (2) , (4), (5) from the comment above also apply.

yes 
>    In inputContentResouceInfo, you declared as dataformat http://w3id.org/meta-share/omtd-share/Token</ns0:annotationType and http://w3id.org/meta-share/omtd-share/PartOfSpeech. That means that the input is something already annotated at that level, correct?

Yes it requires token and pos and lemma.

  >  Is your output a single document for the whole corpus with the extracted terms or annotations of the terms within the corpus?

No, in the output I generate a document for each input document,  so input is corpus and output is corpus

**Comment by gkirtzou** [comment]
About Babelnet:

>> (2) In distributionInfo, change the command to "openminted_freeling process.sh" , or "process.sh" 

Now that I look again at the initial descriptor, I understood that  openminted_freeling is the name of the image, correct? If that's the case, the command should be just "process.sh"

> > (3)
> > In parameterInfo for the parameter "Language", please add a default value as it is obligatory. It
> > would be really helpful when one tries to use it.
> 
> ok, I will put the "auto" value, it is not the most efficient but the most robust one.
> But, the default value is a value that do you provide, or I must also consider that maybe you don't specify any value and I have to deal with that?

It is good practice for obligatory parameters to have a default value. This way, one could use your component without setting anything. In case one want to change the parameter, this could be done via the workflow editor. OMTD platform will complete in the command that is generated in order to run the docker image with the obligatory parameters and values, either with the default value or the provided on from the editor (It will be support soon). So your component will always receive values for the obligatory parameters.

> 
>  > (6)
>   >  In inputContentResourceInfo, you have declared the following data format:
>   >  < ns0:annotationType>http://w3id.org/meta-share/omtd-share/StructuralAnnotationType</ns0:annotationType>
>   > That means that you required as input something that is already annotated at that level, correct?
> 
> In that case I only need text, I don't need nothing else, but once I clicked the option there was no way to remove it. I think it should be empty

Yes, if you complete the metadata from the editor, you cannot remove it. We are working to fix this problem. So in the next version, just don't add the annotation :)

About (5) I will come back to you. 

About the Babelnet component, I think we are good if you make the changes.

**Comment by gkirtzou** [comment]
About (5) You could just need to declare the data format as XMI, so the produced metadata element would look like this

< ns0:dataFormats>
< ns0:dataFormatInfo>
< ns0:dataFormat>http://w3id.org/meta-share/omtd-share/Xmi</ns0:dataFormat>
</ns0:dataFormatInfo>
</ns0:dataFormats>

I wonder why you couldn't find it in the editor. I was able, it is under "XML" and "Document Format". Please let me know if you have other problems. Also, when you update the metadata, please resend them, so I would validate them. 

Thanks

**Comment by joancf** [comment]
Freeling.
i hope that everything is fine now,
I have uploaded a new version of docker and created the new component in the platform
(I don't know how to delete the old one)

Here I atach a test file.
[FreelingInput.zip](https://github.com/openminted/Open-Call-Discussions/files/1927493/FreelingInput.zip)

**Comment by gkirtzou** [comment]
Could you provide the landing page for the newly register component?
Could you also upload a sample input for the Babelnet component?

**Comment by joancf** [comment]
yes...
this is the old one
https://test.openminted.eu/landingPage/component/23a19913-8640-49bf-bb54-de9a588ed800

and that one the new one...
https://test.openminted.eu/landingPage/component/89ba0b71-f878-4d5f-9ec6-577363548257
and I'm still working on Babelnet to upload the new image, but the docker has some GB and takes some time to build and to upload
I'll post a messaga ASAP

**Comment by joancf** [comment]
Bablenet:
The landing page of the new component is
https://test.openminted.eu/landingPage/component/54b0a780-ca02-45d2-98c4-3f04a1abea17
and also here you can find the test files
[babelnetInput.zip](https://github.com/openminted/Open-Call-Discussions/files/1927808/babelnetInput.zip)

I'm uploading the Docker now, and it will take a while,

**Comment by gkirtzou** [comment]
The Freeling metadata record seems correct. I will try to test it using the corpus you provided me. I'll let you know how it goes. 

Similarly, the bablenet metadata record seems correct. I could pull the image, but maybe that happens because uploading hasn't finished from your side.

**Comment by joancf** [comment]
Katerina.
 the BabelNet image now is uploaded,

**Comment by galanisd** [comment]
Hello @joancf 

The Freeling image has been updated?
I tested taln/openminted_freeling:1.0 and I got 
/srv/galaxy/database/jobs_directory/000/714/tool_script.sh: line 9: process.sh: command not found

**Comment by joancf** [comment]
Yes there was an error, I have updated the docker image (with the same tag) 
I think that everything should run fine

**Comment by galanisd** [comment]
Container started....got.

```
INPUT = 'tmp/*.xmi'
OUTPUT = /srv/galaxy/database/jobs_directory/000/867/working/out/
LANG = auto
/bin/process.sh: line 38: cd: UIMA: No such file or directory
/bin/process.sh: line 39: classPath.txt: No such file or directory
Error: Could not find or load main class edu.upf.taln.uima.freeling.FreelingXMIReaderWriter
```

I opened the image (actually ran a cat command in a container).
The last lines of your executable are

`export LD_LIBRARY_PATH=/usr/local/share/freeling/APIs/java/
cd UIMA
export CLASSPATH="target/FreeLingWrapper-0.1-SNAPSHOT.jar":$(<classPath.txt)
java -Xmx450m -cp $CLASSPATH  edu.upf.taln.uima.freeling.FreelingXMIReaderWriter ${INPUT} ${OUTPUT}  ${LANG} xmi`

Comments:

- Exports I assume that should be done in dockerfile..so the image is appropriately configured.
- Galaxy calls process from a directory that it created so this is why "cd UIMA" fails. Use a full path instead.
- -Xmx450m -> you want only 450 MB?

Please can also send your dockefile?

**Comment by joancf** [comment]
I can change the cd UIMA  to cd /UIMA , that is easy ;-) 

The dockerfile is in github.. https://github.com/TalnUPF/OpenMinted_Freeling

But I have another doubt, before doing that change: You make the call with 
INPUT = 'tmp/*.xmi' 
instead of 
INPUT = '/tmp'

I think the input/output path should be absolute. And I read all the xmi files in the folder, I do that because i also expect the typesystem.xml in that folder, and don't split the name into parts

I do not know wat Galaxy does, but a docker run with this command works

docker run -v /var/data/openminted/input:/input -ti -v /var/data/openminted/output:/output openminted_freeling process.sh --input /input --output /output --param:language=en

about the memory, i can increase it ;-)  for me is ok . I did some tests with "auto" loading all the languages and with that memory it can load all the parsers/tokenizers and process a sample file, to be sure that with more files it works I can put 550m

I can't create the classpath in the dockerfile, i can create the file that contains the classpath but I was not able to to do the export  using that file

**Comment by galanisd** [comment]
> `INPUT  = 'tmp/*.xmi'`. 

This is probably our fault. We have to re-create the workflow with which we were testing your  component.

> about the memory, i can increase it ;-)

Usually java allocates more than 450M by default. So, I just asked to be sure that it is not a typo. Put something that is appropriate for your component. 

> I can't create the classpath in the dockerfile, i can create the file that contains the classpath but I was not able to to do the export using that file

Ok keep them as they are now.

> I can change the cd UIMA to cd /UIMA , that is easy ;-)

I had a look in https://github.com/TalnUPF/OpenMinted_Freeling/blob/master/process.sh again.
The important is everything to be Ok when 
`java -Xmx450m -cp $CLASSPATH  edu.upf.taln.uima.freeling.FreelingXMIReaderWriter ${INPUT} ${OUTPUT}  ${LANG} xmi`
is called.
The safest thing to do is keep the  current dir before doing cd /UIMA
`dir=$(pwd)`
then before
java -Xmx450m -cp $CLASSPATH  edu.upf.taln.uima.freeling.FreelingXMIReaderWriter ${INPUT} ${OUTPUT}  ${LANG} xmi

you can return to the dir that Galaxy works by doing
`cd $dir`

So at this point everything will be OK. CLASSPATH  is set and points to your jar files.
One jar contains the class edu.upf....
INPUT OUTPUT and LANG are Ok.

Correct?
So when you are ready please re-register your component so that we can retest.

PS: process.sh is now runnable from any path in the container.

**Comment by joancf** [comment]
Ok you have a new docker, now it runs from any place. and I don't change the folder

I haven't reregistered the component, just updated the docker. I think I'm still in 1.0

**Comment by joancf** [comment]
If it's ok I'll do the same changes in the babelnet component

**Comment by galanisd** [comment]
Reregistered your component and created a workflow (omtdImport -> PDFReader -> Freeling). 
Tested with a corpus and got.

```
INPUT = 'tmp/*.xmi'
OUTPUT = /srv/galaxy/database/jobs_directory/001/1004/working/out/
LANG = auto
xmi second param 2 -auto-
log4j:WARN No appenders could be found for logger (org.apache.uima.resource.metadata.TypeSystemDescription).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Apr 25, 2018 1:08:46 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(411)
INFO: Scanning [file:/working/'tmp/]
Apr 25, 2018 1:08:46 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(237)
INFO: Found [0] resources to be read
Apr 25, 2018 1:08:47 PM edu.upf.taln.uima.freeling.FreeLingWrapper initialize(173)
INFO: Freeling, autodetect mode: true
```

Good news: We are able to call your component.

Bad news: No XMIs were found even though the previous component generated XMI files. It seems that it has to do with the 
`INPUT = 'tmp/*.xmi'`
tmp is a directory that is created were all input data are copied. We do that in all tools
and so far we didn't have any issues.

I had a look in this code (starting at line 84)
https://github.com/TalnUPF/OpenMinted_Freeling/blob/master/src/main/java/edu/upf/taln/uima/freeling/FreelingXMIReaderWriter.java#L84

I am not sure that the XMI reader is configured appropriately. 
Have you tested it?

In case it helps I am doing something similar here:
https://github.com/openminted/omtd-component-executor/blob/master/omtd-component-uima/src/main/java/eu/openminted/workflows/uima/executor/UIMAFitRunner.java#L88

Again an XMI Reader is used which has been tested a million times and it works.

What is also weird is that it prints 
`INFO: Scanning [file:/working/'tmp/]`
why there is a single quote before tmp

**Comment by joancf** [comment]
Ups... solved. 
Please check the new docker it should run (it  did it on my computer)

**Comment by galanisd** [comment]
New error.

```
INPUT = tmp/*.xmi
OUTPUT = /srv/galaxy/database/jobs_directory/001/1041/working/out/
LANG = auto
xmi second param 2 -tmp/od______1271..f0326a791a327c607e519e256f074178.pdf.xmi-
log4j:WARN No appenders could be found for logger (org.apache.uima.resource.metadata.TypeSystemDescription).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Apr 26, 2018 9:36:04 AM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(411)
INFO: Scanning [file:/working/tmp/core_ac_uk__..1db0ba7c3916ec5acda83bbb35edbf56.pdf.xmi]
Apr 26, 2018 9:36:04 AM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(237)
INFO: Found [1] resources to be read
Apr 26, 2018 9:36:05 AM edu.upf.taln.uima.freeling.FreeLingWrapper initialize(173)
INFO: Freeling, autodetect mode: false
java.io.FileNotFoundException: /usr/local/share/freeling/config/tmp/od______1271..f0326a791a327c607e519e256f074178.pdf.xmi.cfg (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at edu.upf.taln.uima.freeling.FreeLingWrapper.init(FreeLingWrapper.java:199)
        at edu.upf.taln.uima.freeling.FreeLingWrapper.initialize(FreeLingWrapper.java:177)
        at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
        at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
        at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
        at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
        at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
        at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
        at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
        at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
        at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
        at org.apache.uima.fit.pipeline.SimplePipeline.runPipeline(SimplePipeline.java:166)
        at edu.upf.taln.uima.freeling.FreelingXMIReaderWriter.main(FreelingXMIReaderWriter.java:98)
Exception in thread "main" org.apache.uima.resource.ResourceInitializationException: Initialization of annotator class "edu.upf.taln.uima.freeling.FreeLingWrapper" failed.  (Descriptor: <unknown>)
        at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:274)
        at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
        at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
        at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
        at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
        at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
        at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
        at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
        at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
        at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
        at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
        at org.apache.uima.fit.pipeline.SimplePipeline.runPipeline(SimplePipeline.java:166)
        at edu.upf.taln.uima.freeling.FreelingXMIReaderWriter.main(FreelingXMIReaderWriter.java:98)
Caused by: org.apache.uima.resource.ResourceInitializationException
        at edu.upf.taln.uima.freeling.FreeLingWrapper.initialize(FreeLingWrapper.java:181)
        at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
        ... 16 more
```

Attached the output of the previous step.
[PdfReader_output.zip](https://github.com/openminted/Open-Call-Discussions/files/1950595/PdfReader_output.zip)

Please check also with this.

**Comment by galanisd** [comment]
- IMO it would make things easier to pass the folder name in FreelingXMIReaderWriter  (e.g. tmp) and not "tmp/*.xmi".
Then in FreelingXMIReaderWriter you can add *.xmi. somehow.

- Is it possible in FreelingXMIReaderWriter to add log messages that print
the args that are you are passing?
E.g.

```
public static void main(String[] args) {
int i = 0;
System. out.println("Command-line arguments:");
for (String arg : args) {
System. out.println(i + "|" + arg + "|");
i++
}
```

It will make debugging easier.

**Comment by galanisd** [comment]
Hi,

any updates?

**Comment by joancf** [comment]
I was checking the system with your long files
Freeling crashes, and I don't think that is a memory issue, and is out of the scope of the integration itself. 

`/Freeling/src/libtreeler/./treeler/srl/paths-container.h:222: void treeler::srl::PathsContainer::ComputeAllPaths(int, int, const treeler::DepVector<std::__cxx11::basic_string<char> >&, const treeler::srl::Children&, const std::__cxx11::list<int>&, bool): Assertion 'prev_path.size() <= 100' failed. `
`

**Comment by galanisd** [comment]
- From the log that I sent you  (`xmi second param 2 -tmp/od______1271..f0326a791a327c607e519e256f074178.pdf.xmi-`) it seems that the parameters are not passed correctly to the jar. Correct? I was not sure...Or this is OK now.

- The "handling of "long" files seems to be another issue... Correct?
If FreeLing can't handle such files then a solution is  to 
shorten names, process the files and either return the output files as they are (with shortened names)
OR restore the original names before your executor finishes; you can keep a mapping somewhere.

**Comment by joancf** [comment]
Yes I have solved the issues with the parameters, 
the problem with you sample files, is not the name (this is something that does not get into FreeLing) the problem is that the file contains a long text extracted from a pdf and it has two problems:
1 paragraphs are not correct (but this is not the main problem)
2 It has strange characters on it, that make freeling crash .
Eclipse has some problems to display the text of your xmi files. 
First I thought that there was a memory problem, but expanding the memory fails at the same point, a sentence which is part of a table with strange tabs
So maybe you can test it with other input files.
I have upload a new version, java can now take up  to 1Gb of menory , but FreeLing is not able to deal with your file :-(

**Comment by greenwoodma** [comment]
The XMI files should be utf-8 encoded are you opening them using a utf-8
aware method?

**Comment by joancf** [comment]
The program considers everything ut-8 and eclipse also indicates that is utf-8
you can try to open the narcis... file on eclipse and check paragraph 1500. It has a very strange behaviour. But in any case I don't know why Freeling crashes, with that text, it is not a problem of memory or size, it's something on the text that makes Freeling crash. 
You have a new version available now

**Comment by galanisd** [comment]
We have tested with this corpus several OpenMinteD tools/web services that are coming either from OpenCall or from OpenMinteD partners. So far, we didn't have any issues. I will try also with another  corpus to see what happens.

Do you have a corpus of XMIs which the Freeling component can process without issues.

**Comment by joancf** [comment]
[input.zip](https://github.com/openminted/Open-Call-Discussions/files/1970455/input.zip)

I attach an input that  can be processed, are two full books (Animal Farm and The Condition of the Working Class in England) 
It takes more than 4 hours to process the second one (on a laptop) but does not fail
I have uploaded a new version of the docker.

**Comment by galanisd** [comment]
- Pulled the latest version of taln/openminted_freeling:1.0

- Created a corpus with just the typesystem and the Animal Farm XMI (B_AnimalFarm.xmi). I didn't want to wait 4 hours.

- Created a worfklow omtdImporter -> freeling.

- Freeling component have started.
Container log:

```
docker logs d7eb3ab538db
INPUT = tmp
OUTPUT = /srv/galaxy/database/jobs_directory/001/1222/working/out/
LANG = auto
xmi second param 2 -auto-
log4j:WARN No appenders could be found for logger (org.apache.uima.resource.metadata.TypeSystemDescription).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
May 03, 2018 1:40:20 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(411)
INFO: Scanning [file:/working/tmp/]
May 03, 2018 1:40:20 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(237)
INFO: Found [1] resources to be read
May 03, 2018 1:40:21 PM edu.upf.taln.uima.freeling.FreeLingWrapper initialize(173)
INFO: Freeling, autodetect mode: true
May 03, 2018 1:40:51 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(336)
INFO: 0 of 1: file:/working/tmp/B_AnimalFarm.xmi
May 03, 2018 1:41:06 PM edu.upf.taln.uima.freeling.FreeLingWrapper process(281)
INFO: Freeleing, the language detected for document is: en
May 03, 2018 1:41:13 PM edu.upf.taln.uima.freeling.FreeLingWrapper init(249)
INFO: Freeling initating Treeler parser for en
May 03, 2018 1:41:57 PM edu.upf.taln.uima.freeling.FreeLingWrapper process(301)
INFO:  start processing from0 size 1686
```

Seems ok so far. Correct? I am waiting to be completed. How much time it takes when you  just run
B_AnimalFarm.xmi?

**Comment by galanisd** [comment]
Completed:

![freeling](https://user-images.githubusercontent.com/10739759/39585220-f004d258-4efc-11e8-8611-37c8f83dc0f5.png)

Output:

[d4f1e1c8-ef7f-4999-b256-9fe84599e855.zip](https://github.com/openminted/Open-Call-Discussions/files/1971475/d4f1e1c8-ef7f-4999-b256-9fe84599e855.zip)

Please can you have a look?

**Comment by joancf** [comment]
The ouput now is correct perfect! I uploaded a new version of the docker whith some minor changes (and tested it)

About the Babelnet. There is a new version with all the problems solved in Freeeling also solved there. You can  take the ouput of Freeling as input. (also it takes time ) or use the input I sent to you. Remember that in any case it needs sentence, token and POS

**Comment by gkirtzou** [comment]
@joancf just to be sure this is the image you want to test taln/openminted_babelnet:1.0, correct? 
And the input you want to test is this one, right?
[BabelnetCorpus.zip](https://github.com/openminted/Open-Call-Discussions/files/1976416/BabelnetCorpus.zip)

**Comment by joancf** [comment]
yes,
I hope that this one is more straightforward :-)

**Comment by joancf** [comment]
also you can use the output from freeling
but it takes a while...

**Comment by gkirtzou** [comment]
I will try both :)

**Comment by gkirtzou** [comment]
Hey @joancf I ran the babelnet component with the input you have provides, and it run successfully
Below is the log and in the attachement the output I got. Could you check that it is what you are expecting?
```

INPUT = tmp
OUTPUT = /srv/galaxy/database/jobs_directory/001/1263/working/out/
May 05, 2018 6:41:31 PM edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter main
INFO: typesystem = /working/tmp/TypeSystem.xml
log4j:WARN No appenders could be found for logger (org.springframework.core.io.support.PathMatchingResourcePatternResolver).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
May 05, 2018 6:41:34 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(412)
INFO: Scanning [file:/working/tmp/]
May 05, 2018 6:41:34 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(238)
INFO: Found [1] resources to be read
May 05, 2018 6:41:36 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(337)
INFO: 0 of 1: file:/working/tmp/en-test.txt.xmi
May 05, 2018 6:41:36 PM edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter main
INFO: language detection, language set to: en
May 05, 2018 6:41:37 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(412)
INFO: Scanning [file:/working/tmp/]
May 05, 2018 6:41:37 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(238)
INFO: Found [1] resources to be read
/UIMA/src/main/resources/config
May 05, 2018 6:41:41 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(337)
INFO: 0 of 1: file:/working/tmp/en-test.txt.xmi

```
[8b6f0425-a013-41d3-9f73-14025903389b.zip](https://github.com/openminted/Open-Call-Discussions/files/1976956/8b6f0425-a013-41d3-9f73-14025903389b.zip)

**Comment by joancf** [comment]
Yes, the results are the ones expected

Joan

**Comment by gkirtzou** [comment]
Perfect. I will run tomorrow a complete workflow with both feeling and
babelnet, just to be sure.

**Comment by gkirtzou** [comment]
I also created a workflow with freeling as step one and babelnet as step two. I fed the workflow with the freeling corpus you have provided me and this is the output I got. Could you check that it is ok? 
[FreelingBabelNetOutputCorpus.zip](https://github.com/openminted/Open-Call-Discussions/files/1977554/FreelingBabelNetOutputCorpus.zip)

**Comment by joancf** [comment]
Yes the result is the the expected one, it is correct! 
but only a portion of the Animal's farm is there, the first paragraphs

**Comment by gkirtzou** [comment]
I ran this file that you have uploaded in the first days of our discussion. I hope that it is the same with the processed (apart from the annotations).
[FreelingCorpus.zip](https://github.com/openminted/Open-Call-Discussions/files/1977616/FreelingCorpus.zip)

**Comment by joancf** [comment]
Sorry, Yes, you are right! I think that now everything is correct!

**Comment by gkirtzou** [comment]
Perfect! I just tried another small test where I wanted to test both freeling + babelnet with this small corpus https://test.openminted.eu/landingPage/corpus/97833edb-b6c7-44a0-9d8d-f9219a147e2a, a typical corpus found in OMTD platform. I first user tika converter to transform pdf file to xmi (with DKPro typesystem), and then passed them to freeling, but the freeling failed to process the files.

Below is the log

```

INPUT = tmp
OUTPUT = /srv/galaxy/database/jobs_directory/001/1281/working/out/
LANG = auto
xmi second param 2 -auto-
log4j:WARN No appenders could be found for logger (org.apache.uima.resource.metadata.TypeSystemDescription).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
May 07, 2018 12:39:50 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(411)
INFO: Scanning [file:/working/tmp/]
May 07, 2018 12:39:50 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(237)
INFO: Found [2] resources to be read
May 07, 2018 12:39:51 PM edu.upf.taln.uima.freeling.FreeLingWrapper initialize(173)
INFO: Freeling, autodetect mode: true
May 07, 2018 12:40:19 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(336)
INFO: 0 of 2: file:/working/tmp/41398_2017_Article_78.pdf.xmi
May 07, 2018 12:40:23 PM edu.upf.taln.uima.freeling.FreeLingWrapper process(278)
SEVERE: Freeling, error in language detection, skip document
May 07, 2018 12:40:24 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(336)
INFO: 1 of 2 (50%  ETA 00:00:33.438  RUN 00:00:33.438  AVG 33438  LAST 4491): file:/working/tmp/fphar-09-00130.pdf.xmi
May 07, 2018 12:40:27 PM edu.upf.taln.uima.freeling.FreeLingWrapper process(278)
SEVERE: Freeling, error in language detection, skip document
```

**Comment by joancf** [comment]
I see that the language detection fails, have you tried to set the language to "en" instead of auto. Also I think that FreeLing has problems with pdf's transformed to text when the text is very dirty.

**Comment by gkirtzou** [comment]
Yes the language was set to auto, I will try with english ("en") to see if that would be better.

**Comment by gkirtzou** [comment]
Ok, I reran the worfklow (omtdImporter + tika converter + freeling (en) + babelnet))  with this corpus (https://test.openminted.eu/landingPage/corpus/97833edb-b6c7-44a0-9d8d-f9219a147e2a) , but this time the babelnet failed

```
INPUT = tmp
OUTPUT = /srv/galaxy/database/jobs_directory/001/1303/working/out/
May 07, 2018 2:06:41 PM edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter main
INFO: typesystem = /working/tmp/TypeSystem.xml
log4j:WARN No appenders could be found for logger (org.springframework.core.io.support.PathMatchingResourcePatternResolver).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
May 07, 2018 2:06:43 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(412)
INFO: Scanning [file:/working/tmp/]
May 07, 2018 2:06:43 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(238)
INFO: Found [2] resources to be read
May 07, 2018 2:06:44 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase nextFile(337)
INFO: 0 of 2: file:/working/tmp/41398_2017_Article_78.pdf.xmi
May 07, 2018 2:06:45 PM edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter main
INFO: language detection, language set to: x-unspecified
May 07, 2018 2:06:46 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase scan(412)
INFO: Scanning [file:/working/tmp/]
May 07, 2018 2:06:46 PM de.tudarmstadt.ukp.dkpro.core.api.io.ResourceCollectionReaderBase initialize(238)
INFO: Found [2] resources to be read
Exception in thread "main" org.apache.uima.resource.ResourceInitializationException: Unexpected Exception thrown when initializing Custom Resource "edu.upf.taln.uima.babelnet.BabelnetSenseInventoryResource" from descriptor "<unknown>".
	at org.apache.uima.impl.CustomResourceFactory_impl.produceResource(CustomResourceFactory_impl.java:96)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:248)
	at org.apache.uima.resource.impl.ResourceManager_impl.registerResource(ResourceManager_impl.java:720)
	at org.apache.uima.resource.impl.ResourceManager_impl.initializeExternalResources(ResourceManager_impl.java:594)
	at org.apache.uima.resource.Resource_ImplBase.initialize(Resource_ImplBase.java:210)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.initialize(AnalysisEngineImplBase.java:157)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:133)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
	at org.apache.uima.fit.pipeline.SimplePipeline.runPipeline(SimplePipeline.java:166)
	at edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter.main(BabelNetXMIReaderWriter.java:90)
Caused by: java.lang.IllegalArgumentException: Errors initializing [class edu.upf.taln.uima.babelnet.BabelnetSenseInventoryResource]
Failed to convert property value of type 'java.lang.String' to required type 'it.uniroma1.lcl.jlt.util.Language' for property 'babelNetLang'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [java.lang.String] to required type [it.uniroma1.lcl.jlt.util.Language] for property 'babelNetLang': no matching editors or conversion strategy found
Failed to convert property value of type 'java.lang.String' to required type 'it.uniroma1.lcl.jlt.util.Language' for property 'babelNetDescLang'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [java.lang.String] to required type [it.uniroma1.lcl.jlt.util.Language] for property 'babelNetDescLang': no matching editors or conversion strategy found
	at org.apache.uima.fit.component.initialize.ConfigurationParameterInitializer.initialize(ConfigurationParameterInitializer.java:179)
	at org.apache.uima.fit.component.initialize.ConfigurationParameterInitializer.initialize(ConfigurationParameterInitializer.java:203)
	at org.apache.uima.fit.component.initialize.ConfigurationParameterInitializer.initialize(ConfigurationParameterInitializer.java:219)
	at org.apache.uima.fit.component.Resource_ImplBase.initialize(Resource_ImplBase.java:57)
	at edu.upf.taln.uima.babelnet.BabelnetSenseInventoryResource.initialize(BabelnetSenseInventoryResource.java:79)
	at org.apache.uima.impl.CustomResourceFactory_impl.produceResource(CustomResourceFactory_impl.java:92)
	... 23 more
```
It would be good to be able to handler xmi that are generated from pdf /xml files, as this is the main corpora available in the OMTD platform.

**Comment by gkirtzou** [comment]
I am also attaching the intermediate results, that could be of help from tika and freeling
[tika_output.zip](https://github.com/openminted/Open-Call-Discussions/files/1980261/tika_output.zip)
[freeling_output.zip](https://github.com/openminted/Open-Call-Discussions/files/1980263/freeling_output.zip)

**Comment by joancf** [comment]
Yes it would ;-) 
I can explain why this happens, 
If you say to the Freeling wrapper  that the language is "en" it will analyze the document without checking, nor setting, the document language . So if your document does not specify a any language 
Freeling only sets it in auto mode
When it gets into the the Babelnet component, that component expects the language to be defined in the document, and as it does not find the language, it fails.
I'm not sure the best way to put the language in the document. FreeLing should do that, but I think it gets confused with some dirty parts of the text
 I think that before getting into the pipeline the pdf's need some cleaning (that used to happen to us with other kind of documents) tables,formulas....become very strange for parsers and other tools that do not behave as expected.
Putting the language to input document everything should run smoothly.

**Comment by gkirtzou** [comment]
@joancf  Ok, I now understand what the problem is. Maybe it would be a good idea if after the conversion from pdf or xml to xmi there was a component that would clean up the xmi by removing possible areas that would generate errors. I don't know if such a component exists within OMTD platform. 

Anyway, both your components can successfully run in the platform, thus testing has been completed. It only remains to upload them to the production site. I will let you know, when we will be ready to do this.

**Comment by joancf** [comment]
Hi, 
i did the following
1 edited your tika_output files to add "en" as language  (tika_input)
2 processed with freeling  (Freeling_output)
3 processed with babelnet (babelnet_output)
[tika_en.zip](https://github.com/openminted/Open-Call-Discussions/files/1983180/tika_en.zip)
here you have the input and outputs, you should be able to do the same processing. (but files may not be 100% equal as there is no guarantee  that for the same content the xmi will be the same)

**Comment by joancf** [comment]
Ok, we have been crossing comments

**Comment by gkirtzou** [comment]
@joancf Sorry for putting you into extra trouble.

**Comment by reckart** [comment]
The error below indicates that uimaFIT does not know how to convert the value "en" for the string parameter language into an instance of `it.uniroma1.lcl.jlt.util.Language`.

```
Caused by: java.lang.IllegalArgumentException: Errors initializing [class edu.upf.taln.uima.babelnet.BabelnetSenseInventoryResource]
Failed to convert property value of type 'java.lang.String' to required type 'it.uniroma1.lcl.jlt.util.Language' for property 'babelNetLang'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [java.lang.String] to required type [it.uniroma1.lcl.jlt.util.Language] for property 'babelNetLang': no matching editors or conversion strategy found
Failed to convert property value of type 'java.lang.String' to required type 'it.uniroma1.lcl.jlt.util.Language' for property 'babelNetDescLang'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [java.lang.String] to required type [it.uniroma1.lcl.jlt.util.Language] for property 'babelNetDescLang': no matching editors or conversion strategy found
	at org.apache.uima.fit.component.initialize.ConfigurationParameterInitializer.initialize(ConfigurationParameterInitializer.java:179)
```

Please change the parameter in your component to be a simple string parameter such as:

```
    /**
     * Use this language instead of the document language to resolve the model.
     */
    public static final String PARAM_LANGUAGE = ComponentParameters.PARAM_LANGUAGE;
    @ConfigurationParameter(name = PARAM_LANGUAGE, mandatory = false)
    protected String language;
```

Then overwrite/implement the `initialize` method of the component and manually instantiate the `it.uniroma1.lcl.jlt.util.Language` from the language parameter value. Mind to call `super.initialize` when you overwrite `initialize`.

**Comment by reckart** [comment]
```
May 07, 2018 12:40:27 PM edu.upf.taln.uima.freeling.FreeLingWrapper process(278)
SEVERE: Freeling, error in language detection, skip document
```

Text extracted from PDF files is always pretty dirty. I expect that your language detector does not actually look at the entire document but probably only at the first part of the document. In scientific publications, this first part includes the title and author information. In particular the latter is not well suited for language detection.

Please check if you could improve your language detection component by skipping into later parts of the document. You might want to use a simple tokenization/sentence detection mechanism and search for sentences which actually look as such (e.g. starting with a capital letter, ending with a full stop, being somewhere between 6 and 20 words long). These should be much better suited to perform language detection.

**Comment by joancf** [comment]
@reckart 
about the Babelnet error. If the language is Set in the document then it works the problem was thatthe document had as language, "x-unspecified" which does not correspond to any known language. 

About language detection. I have tried to find the end of a sentence after the first 800 characters (150-200 words). I could detect it, with no problem, but the text is crazy, and FreeLing fails to detect, 
There are Tika errors, the authors are Xinese and there are a lot of very specific terms. I think that the number of English "normal" words in this part is a very small subset  I attach the selected texts ...
So I think can't do more on this. I have uploaded a new version of the component with this new feature.

This is the selected text for the first article 
`
ART ICLE Open Ac ce s s

Metabolite identification in fecal
microbiota transplantation mouse livers
and combined proteomics with chronic
unpredictive mild stress mouse livers
Bo Li1,2,3, Kenan Guo4, Li Zeng1,2, Benhua Zeng4, Ran Huo1,2,3, Yuanyuan Luo1,2,5, Haiyang Wang1,2, Meixue Dong1,2,
Peng Zheng1,2,6, Chanjuan Zhou1,2, Jianjun Chen1,2, Yiyun Liu1,2, Zhao Liu1,2, Liang Fang5, Hong Wei2 and
Peng Xie2,3,5,6,7

Abstract
Major depressive disorder (MDD) is a common mood disorder. Gut microbiota may be involved in the pathogenesis of
depression via the microbe?gut?brain axis. Liver is vulnerable to exposure of bacterial products translocated from the
gut via the portal vein and may be involved in the axis.
`

The second article it only selects elements from the header, and with some missing characters ?
`
fphar-09-00130 February 21, 2018 Time: 17:55 # 1

ORIGINAL RESEARCH
published: 22 February 2018

doi: 10.3389/fphar.2018.00130

Edited by:
Judith Maria Rollinger,

University of Vienna, Austria

Reviewed by:
Marinella De Leo,

Universit? degli Studi di Pisa, Italy
Xiaoying Zhang,

Northwest A&F University, China
Marc Poirot,

Institut National de la Sant? et de la
Recherche M?dicale (INSERM),

France

*Correspondence:
Qingwang Liu

liuqingwang312@163.com
Jie Yuan

dsyuanjie@126.com

Specialty section:
This article was submitted to

Ethnopharmacology,
a section of the journal

Frontiers in Pharmacology

Received: 29 November 2017
Accepted: 06 February 2018
Published: 22 February 2018

Citation:
Zhang Z, Fang T, Zhou H, Yuan J
and Liu Q (2018) Characterization

of the in Vitro Metabolic Profile
of Evodiamine in Human Liver

Microsomes and Hepatocytes by
UHPLC-Q Exactive Mass

Spectrometer

`

**Comment by gkirtzou** [comment]
Dear @joancf  you can now proceed to the uploading of your applications at https://services.openminted.eu/home

Just, some final suggestions, not obligatory but recommended, for the metadata records are 
- Add in both input and output resource info the language(s) that your BabelNet component can handle.
-  Complete the documentation info, with user manuals & publication for citation, in both your components. 
- Add the resourceCreator with contact with your self, for citation reasons, in both your components. 

Please, when you upload your components, create the appropriate workflows so that someone could run your component using the workflow editor. For more info see https://openminted.github.io/releases/workflow-editor/ 
Please also let me know when you have uploaded the applications to the production site.
If you encounter any problems, please let us know.
Thanks

**Comment by joancf** [comment]
ok I'll do, but...
about languages and Babelnet, Babelnet has 271 languages and I don't have the list!

**Comment by gkirtzou** [comment]
@joancf Really? It can work in 271 languages? Impressive! Anyway, the language element is not obligatory, it is just a recommendation. If it is too difficult, you can just skip my recommendation.

**Comment by joancf** [comment]
Well it should ;-) . I haven't tested all of them, I have used it in English, Spanish and Arabic texts and it works.  Babelnet is a multilingual resource based on wikipedia and other resources, the coverage changes from language to language

**Comment by gkirtzou** [comment]
Maybe you could add this info, that is a multilingual resource based on wikipedia and other resources,  in the description if you haven't done already.

**Comment by joancf** [comment]
About the manuals. Is there a public access to the deliverables we presented to you? or do we add them in the github repositories?
Joan

**Comment by gkirtzou** [comment]
Note that when you create an application, you wiil be asked to fill in a metadata record. Some tips for filling it in - so that they are discoverable by the users but also that users can cite you and your resource. 
- Give the application a unique name that humans can read and short enough
- Give an explanatory description; remember you can re-use the description of the component(s) in the workflow accumulatively
-  Add in the inputProcessingResourceInfo the information of the first component used in the workflow (the one after omtdImpoter), and in the outputResourceInfo the information of the last component; if the annotations from previous components are retained in the final output, please add those as well
-  (optionally) You can use the relations set with the relation "hasPart" to document the components used in the workflow - it can be repeated multiple times.

About the deliverable I have to ask, I am not sure. You can add the github issue page for example as  onLineHelpURL with identifierSchema=URL

**Comment by gkirtzou** [comment]
Please, when you upload your component and you create an application so that a non-expert user could run it, please keep them private and send me the metadata here, so I could check them.

Also, you could add the deliverable from the tender call to your git-hub repository (if you have) and add them as documentation to the metadata of your software.

Thanks.

**Comment by joancf** [comment]
I have uploaded both components
-https://services.openminted.eu/landingPage/component/6958ed38-4494-4af2-953a-d78c35454257
-https://services.openminted.eu/landingPage/component/42c463b4-2d8b-4125-adc4-93f89918ce09
I can't create a corpus
I can't create a workflow (the components just uploaded do not appear in the list of selectable components)

**Comment by gkirtzou** [comment]
@joancf 
There is an issue with the AAI and for the moment you cannot create a corpus or a workflow. I will let you know, when the issue is resolved. 


> I can't create a workflow (the components just uploaded do not appear in the list of selectable components)

Question what do you mean here? I can see the components in the workflow editor.

**Comment by joancf** [comment]
ok, I was creating an application
also I don't understand why the  workflow indicates: 
component_InputFiles: Data input 'component_InputFiles' (pdf)

**Comment by gkirtzou** [comment]
Where do you see that? I am not sure I understand.

**Comment by joancf** [comment]
when creating a workflow
![image](https://user-images.githubusercontent.com/2848623/40014733-a1742aa4-57b1-11e8-8f97-1fe32eeb1462.png)

**Comment by gkirtzou** [comment]
@joancf Oh I see. Yes, this is a small bug from our side not rendering correctly the input information in the metadata record of the component. For the moment please ignore it.

**Comment by gkirtzou** [comment]
@joancf the problem with had with AAI has been resolved. You can process creating the application(s) so that non tdm expert could use your components. Please not my suggestions above and keep the app(s) private until the are tested and their metadata are approved. When you are ready, please do send me the metadata, so I could check them.

**Comment by joancf** [comment]
ok  @gkirtzou  I did it,
it says running, since yesterday...

**Comment by gkirtzou** [comment]
1) I cannot anything that shows that your components were executed. Could you please tell me how you created your workflow? What components did you use? Maybe a screenshot would be easier.
2) Also could you send me the metadata of your app(s) to check them?

**Comment by joancf** [comment]
![image](https://user-images.githubusercontent.com/2848623/40173225-66e0bbe2-59d1-11e8-99d0-a25116b393c7.png)
https://services.openminted.eu/landingPage/corpus/72b93a62-d82d-45c6-acac-82ab4d32e58c

but the process looks a bit strange
![image](https://user-images.githubusercontent.com/2848623/40173510-666b761a-59d2-11e8-8020-00cb9a7499f5.png)

and this is the workflow
![image](https://user-images.githubusercontent.com/2848623/40173604-ad406122-59d2-11e8-822b-d15a8c53318b.png)

**Comment by gkirtzou** [comment]
Ok the workflow is not according to the OMTD specifications, thus it appears to run, but actually it 
cannot run. 

The first component you are using is not the correct one. You must use the omtdImporter (under DataImport). If I remember correctly, UIMA takes as input XMI, right? If that's the case the you need to use a converter, such as PdfReader to convert data from pdf to xmi. For more information how to correctly create and configure a workflow, please check here : https://openminted.github.io/releases/workflow-editor/1.0.0/workflow


Also, please send me here the xml files for the metadata of your application and the corpus, so I could check them. I cannot open the URIs, since they are private. Don't change them yet, so you can make changes according to my suggestions.

**Comment by gkirtzou** [comment]
@joancf were you able to update the workflow according to the provided specs? Let me know if you need any help.

**Comment by joancf** [comment]
@gkirtzou no couldn't 
I added an impoter, i saved the application, an it says I have an input dataset, i rut and it keeps running.
Can you try it,  with the simple workflow  that you have in the image above ?

**Comment by gkirtzou** [comment]
Where does it say " I have an input dataset," ?

Ok, I will try to make the applications for you and let you know how that goes.

**Comment by gkirtzou** [comment]
Ok I have create an app for the freeling component. In the attachement you can find the metadata of the app and the output. Could you check if they are ok? 
[e1181e93-dcd6-43c5-bf50-da28ba5e7a78.zip](https://github.com/openminted/Open-Call-Discussions/files/2016760/e1181e93-dcd6-43c5-bf50-da28ba5e7a78.zip)
[FreelingApp.xml.zip](https://github.com/openminted/Open-Call-Discussions/files/2016770/FreelingApp.xml.zip)

**Comment by gkirtzou** [comment]
@joancf I talked with a colleague and he told me that we have a language identification in the test platform. In the attachment you would find the results from the following workflow  : pdfReader->LanguageIdentifier->freeling->babelnet

Could you check that they what you expect? If they are ok, i could try to add that component to the service and create a workflow for babelnet as well. 
[babelnet_output_identification.zip](https://github.com/openminted/Open-Call-Discussions/files/2016948/babelnet_output_identification.zip)

**Comment by gkirtzou** [comment]
@joancf Could you validate the results I have sent if is what you are expecting? I need this in order to complete the registration of your component and creating the correct applications.

**Comment by gkirtzou** [comment]
@joancf Could you please validate the results? We need to proceed with the publications of the apps ASAP!

**Comment by joancf** [comment]
@gkirtzou, Kateria (first of all, sorry for the delay, yesterday was Holiday and today I was in a meeting till now)
In any of the files you sent me I can't see the output of the freeling/babelnet processing.

**Comment by joancf** [comment]
So I don't know how to proceed right now. 
For me the simplest way is to have a workflow that accepts xmi, and produces xmi, 
with the xmi files I have for testing. 
maybe they are not correct, I upload here the file I used, 
[fulltext.zip](https://github.com/openminted/Open-Call-Discussions/files/2026240/fulltext.zip)

**Comment by joancf** [comment]
and here is the output I get when processing with Freeling first
[freeling_out.zip](https://github.com/openminted/Open-Call-Discussions/files/2026283/freeling_out.zip)

and BabelNet just next
[Babelnet_out.zip](https://github.com/openminted/Open-Call-Discussions/files/2026284/Babelnet_out.zip)

**Comment by gkirtzou** [comment]
>  In any of the files you sent me I can't see the output of the freeling/babelnet processing.

Xmmm, maybe the files I tested (a small corpus created by the OMTD platform) didn't have the terms you are looking for. I will try with the corpus you send me, but I think it would be nice to be able to run with publications from the omtd platform. Maybe the publications from biological domain is not the right one. Could you think a domain where we could search and build a corpus based on what terms you are extracting?

**Comment by joancf** [comment]
I don't think this is the problem.Because Freeling should generate at least tokens, and it seems it does not do nothing
Do you have the output produced by freeling (or babelnet) when you processed these files?

**Comment by gkirtzou** [comment]
In the attachement you would find the output of the following steps (which is input to the next step)
1) Language Identification
2) Freeling
3) Babelnet
[test2.zip](https://github.com/openminted/Open-Call-Discussions/files/2026384/test2.zip)

**Comment by joancf** [comment]
Hi, I took your data (from language identification) and I could process it with the dockers (the same same version that is uploaded in hub.docker)
the results I obtained are these ones:
Freeling output
[fr_out.zip](https://github.com/openminted/Open-Call-Discussions/files/2027829/fr_out.zip)

Babelnet output
[bl_out.zip](https://github.com/openminted/Open-Call-Discussions/files/2027831/bl_out.zip)
Because  the ones you sent me  don't seem to have the process done!

**Comment by gkirtzou** [comment]
Ok, I see that indeed the results are different. Two thing come to my mind, why the Freeling fails to generate tokens in the input:

1. Whether the language parameter has an effect. So what language have you used for the Freeling component?
2. Whether have you update the image within the last couple of weeks? Because that's when we last pull the image in the infrastructure.

**Comment by joancf** [comment]
I did some changes to manage the language,
I ran the docker with language="en" but now I tried it with language "auto" and with the image uploaded two weeks ago it fails, the current one works.

**Comment by gkirtzou** [comment]
Ok, I had tried with language="auto". I will retry with language="en". I will also repull the image to be on the same page. Note that since your images are big, the platform is not pulling it every time, but only once in the registration. The idea is not to image does not change since registration to verify reproducibility.

**Comment by joancf** [comment]
yes images are huge ;-)  include all the Freeling, but i did not increase the version till it has proven to work

**Comment by gkirtzou** [comment]
Ok, but just let me know that you change the image, because due to size the pulling is done manually. So next time just let me know :-)

**Comment by gkirtzou** [comment]
@joancf I have tested the updated image, with language='en' for the Freeling parameter. In the attachement you can find the results of both steps of the workflow (your components). It looks more similitar with the one you have sent me but, there are some differences (it contain also some extra tags). Could you verify if there as expected?

[MyTest2.zip](https://github.com/openminted/Open-Call-Discussions/files/2030974/MyTest2.zip)

**Comment by joancf** [comment]
Yes, this data is correct!!  :-)

**Comment by gkirtzou** [comment]
Perfect! Then I can create a similar workflow to the services.openminted.eu. I will let you know how it goes.

**Comment by gkirtzou** [comment]
@reckart Could you please upload to the services.openminted.eu the UIMA language identification, so i could create a correct application for @joancf ? Thanks!

**Comment by joancf** [comment]
another option is to check why my files fulltext.zip are not working...

**Comment by gkirtzou** [comment]
> another option is to check why my files fulltext.zip are not working...

What do you mean your fulltext.zip files are not working?

**Comment by joancf** [comment]
I uploaded some test files as an input corpus (the fulltext.zip)
 I did an application, and the application whith that corpus as input, 
The application never ended,

**Comment by gkirtzou** [comment]
@joancf what is the other identifier of your application? (shown at the bottom of the page of the landing page). Also approximately when did you run the app? two hours ago?

**Comment by joancf** [comment]
I run it some days ago, today I tried it again (only once) but it appears that there are several processes running... I don't know... but none of them has ended
![image](https://user-images.githubusercontent.com/2848623/40442048-ff209206-5ec2-11e8-9ae1-bce89ffdf221.png)
these are the Identifiers
    OMTD: 328ef54b-4c6b-4a14-905d-06d2a0e04698 
    OTHER: 0931731504677607-75ba0ebd-c7a2-436d-9f47-8f16ea7d4429 

I ran it, after you said you had uploaded the last docker image

**Comment by gkirtzou** [comment]
@joancf ok the old ones that say "Running" probably have either failed or successed but due to the bug that exists at the platform at the time, the registry has been able to update correctly. I was able to run your component successfully, I wonder why you cannot. Could you send me a screenshot of the workflow from the workflow editor to check its structure? And what data does the freeling_test corpus contain? 

An the mean time, I will try to search for the latest run of your workflow for see the log.

**Comment by gkirtzou** [comment]
For some reason that is not clear to me at all, I cannot find anything in the slaves. It seems like your component never run. Could you please try once again now? Let me know when you do, so I could keep an eye in the logs

as for the multiple instances, probably you pressed the button "Run" more than once, and you didn't even notice.

**Comment by joancf** [comment]
I'll do it in the afternoon now I'm in a meeting

**Comment by joancf** [comment]
I'm late to do it today,  let's do it tomorrow morning, it's ok at 10:00 AM (GMT+2 Barcelona time) for you?

**Comment by gkirtzou** [comment]
Yes please, let's do the testing tomorrow. I also have to leave now.

**Comment by gkirtzou** [comment]
@joancf I am available, let me know when you start a new run.

**Comment by joancf** [comment]
@gkirtzou  Katerina  maybe we can contact by a chat channel while we do it. maybe using skype my user is joancf

**Comment by gkirtzou** [comment]
@joancf  You can find me at skype under gkirtzou

**Comment by gkirtzou** [comment]
@joancf I am trying to create an application using both the freeling and the babelnet component as I did in test.openminted.eu. The application rfinished correctly, but while the babelnet component runs the following exception occurs (note that both components are updated to the latest image of v1.0 with docker pull) :

```

SEVERE: null
java.lang.NullPointerException
	at edu.upf.taln.uima.babelnet.BabelNetCandidateIdentification.process(BabelNetCandidateIdentification.java:133)
	at org.apache.uima.analysis_component.JCasAnnotator_ImplBase.process(JCasAnnotator_ImplBase.java:48)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:401)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.processAndOutputNewCASes(PrimitiveAnalysisEngine_impl.java:318)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.process(AnalysisEngineImplBase.java:269)
	at org.apache.uima.fit.pipeline.SimplePipeline.runPipeline(SimplePipeline.java:176)
	at edu.upf.taln.uima.babelnet.BabelNetXMIReaderWriter.main(BabelNetXMIReaderWriter.java:90)
```
Any idea what could be going wrong? I am attaching the files as given by both components. Could you check if they are ok? 

[taln_results.zip](https://github.com/openminted/Open-Call-Discussions/files/2057916/taln_results.zip)

**Comment by joancf** [comment]
Hi
The files are correct,
about the error, is inside a try/catch so the program continues. It happens because there is a token with no POS tag.

**Comment by gkirtzou** [comment]
@joancf Ok, I got confused for a moment there. Anyway, I create an application that uses both your component, which is available here https://services.openminted.eu/landingPage/application/6a9ce7dc-aa6f-4119-a8f7-5a0e260c2b44, and it runs as expected.

There is nothing else to do from our part. So I am closing the issue.

Cheers
